frontend/src/run.js

```diff
@@ -162,120 +151,95 @@ export class Run extends React.Component {
     return this.props.location.hash !== '' ? this.props.location.hash.substring(1) : defaultValue;
   }
 
-  buildTree(results) {
-    let tests = {
-      '_sub': {
-      },
-      '_stats': {
-        'count': 0,
-        'passed': 0,
-        'failed': 0,
-        'skipped': 0,
-        'error': 0,
-        'xpassed': 0,
-        'xfailed': 0
-      },
-      '_duration': 0.0
-    };
-    results.forEach((testResult) => {
-      this.buildTreeObject(cleanPath(testResult.metadata.fspath) + '/' + testResult.test_id, tests, testResult);
-    });
-    let cloud = [];
-    this.buildJSTreeStructure(tests, cloud);
-    return {core: {data: cloud}};
+  onTreeItemSelect = (event, treeItem) => {
+    if (treeItem && ! treeItem.children) {
+      this.setState({activeItems: [treeItem], testResult: treeItem._testResult});
+    }
   }
 
-  buildTreeObject(path, container, data) {
-    let segs;
-    if (!Array.isArray(path)) {
-      segs = processPyTestPath(path);
-    }
-    else {
-      segs = path;
-    }
-    let head = segs[0];
-    let end = segs.slice(1);
-    if (end.length === 0) {
-      container['_sub'][head] = data;
-      if (Object.prototype.hasOwnProperty.call(container['_stats'], data['result'])) {
-        container['_stats'][data['result']] += 1;
-      }
-      else {
-        container['_stats']['failed'] += 1;
+  buildTree(results) {
+    function getPassPercent(stats) {
+      let percent = 'N/A';
+      if (stats.count > 0) {
+        percent = Math.round(((stats.passed + stats.xfailed) / stats.count * 100));
```

Comment:
I may not have had my coffee today; but I'm surprised you can reassign `percent` a new value here.

Reply:
Never mind, I've simply been away from JS for too long